src/vue-components/lookup.vue

```diff
@@ -42,38 +42,34 @@
       return {
         lookuptext: '',
         showLanguageSettings: false,
-        initLanguage: null,
-        currentLanguage: null,
         options: {},
-        resourceOptions: {}
+        resourceOptions: {},
+
+        currentLanguage: null
       }
     },
     props: {
       uiController: {
         type: Object,
         required: true
-      },
-      parentLanguage: {
-        type: String,
-        required: false
       }
     },
     created: function () {
       if (this.uiController) {
-        this.options = this.uiController.options.clone(TempStorageArea)
-        this.resourceOptions = this.uiController.resourceOptions.clone(TempStorageArea)
-        if (this.parentLanguage) {
-          this.initLanguage = this.parentLanguage
-          this.currentLanguage = this.parentLanguage
-        } else {
-          this.currentLanguage = this.options.items.preferredLanguage.currentTextValue()
+        this.options = this.uiController.options
+        this.resourceOptions = this.uiController.resourceOptions
+
+        console.log(`at creation lookup language is ${this.options.items.lookupLanguage.currentTextValue()}`)
+
+        let vm = this
+        window.onbeforeunload = function (){
+          vm.options.items.lookupLanguage.removeItem()
```

Comment:
I'm wondering why do we need to remove an item on document close. Do I understand correctly that this is a language that was used to for a latest lookup query? Do we not want to save it because it is related to the current session only and we want to start a new user session with clean data?

If lookup language is transient and we don't want to store it permanently, maybe it's better not to save it to storage? For that, we can probably add a `transient` option to the setting definition:
```
"lookupLanguage": {
      ...
      "transient": true,
      ...
    }
```
With this option set, set methods can be modified to never save such option value to the storage. I think it will be clearer and more effective (less writes to the local storage overall).

Or do I not understand it correctly and we do have a different purpose here?

It's also important to note that local storage is shared between all open windows. So if we have one window where lookup language is set and it is stored to the local storage, it will be visible to other windows (unless we use session storage, not local storage). Probably this is not what we want. Maybe this is one more reason not to save it to the local storage.

I also think it's probably better not to use `beforeunload` event. Vue usually does not recommend to access HTML elements/events directly as it might interfere with Vue's internal event handling. If we still need to remove it, probably it's better to use a corresponding Vue's event like `beforeDestroy`.

What do you think?

Reply:
@irina060981 asked me if the setting should be kept across instances or removed when the component is destroyed, and I said the latter.  We need to do work on the setting state and have a finer grained ability to control what is saved where when, something that I think needs to wait for our state refactoring generally. I like the suggestion of having a transient flag on a setting which means it doesn't get saved to storage, but I think we may need even more granular control than that. We need, for example, to be able to distinguish between browser session and "page" session.  

Here's how I think this should work right now:

When the user accesses the lookup component, it should default to whatever language the current language they are using is. So, if they access it from the popup, it should default first to the language of the word that was used to produce the popup.  If they access it from the panel, if the panel is showing the results of a current word lookup, it should default to that language, otherwise it should default to whatever the language identified for the page was (which may end up being the preferredLanguage setting).

So the setting should not be persisted across browser tabs, and instead specific to the individual browser page.

Reply:
OK - I will rethink about the workflow according to this requirement.
About
"I also think it's probably better not to use beforeunload event. Vue usually does not recommend to access HTML elements/events directly as it might interfere with Vue's internal event handling. If we still need to remove it, probably it's better to use a corresponding Vue's event like beforeDestroy."

BeforeDestroy hook executes if the component is destroying but page is not reloaded.
Otherwise this lifecycle hook is not executed.
In our case - the component is embedded in the panel or popup. And they are not ever destroyed - they could become visible and invisible. So in our case I couldn't use beforeDestroy hook (I tried to use at the first but faced with this problem). That is why I don't see any other way to have a chance to do something with component before page is reloaded.